src/dev/build/lib/download.ts

```diff
@@ -83,30 +85,39 @@ export async function downloadToDisk({
       }
 
       const hash = createHash(shaAlgorithm);
+
       await new Promise((resolve, reject) => {
         response.data.on('data', (chunk: Buffer) => {
-          hash.update(chunk);
+          if (!skipChecksumCheck) {
+            hash.update(chunk);
+          }
+
           writeSync(fileHandle, chunk);
         });
 
         response.data.on('error', reject);
         response.data.on('end', resolve);
       });
 
-      const downloadedSha = hash.digest('hex');
-      if (downloadedSha !== shaChecksum) {
-        throw new Error(
-          `Downloaded checksum ${downloadedSha} does not match the expected ${shaAlgorithm} checksum.`
-        );
+      if (!skipChecksumCheck) {
```

Comment:
could some additional checks be done if we are not checking checksums? I think in the issue we saw yesterday the file size was `0`.

Reply:
What about a check where if we've written 0 bytes after the download is complete we throw an error and remove the empty file?